versioned_docs/version-2021sp/lecture7.md

```diff
@@ -7,6 +7,426 @@ Lecture Video will be posted here after lecture
+
 [Final Project Team Formation Form due 4/11 11:59PM (NO SLIP DAYS USABLE)](https://forms.gle/wQnrAwEB44oYNbCa8)
 
 [Final Project (Milestone 0 due Thursday 4/15 3:59PM)](/docs/finalproject)
+
+## Hooks, and More on Functional Components in React
+
+React has changed a lot in the past few years since it was released in 2013,
+in true JavaScript library fashion. A major new concept leading the way of much
+innovation for React development is Hooks, released in **2019(!)**. While there
+exist many Hooks in React (and you can even develop your own), we will be focusing
+on two main Hooks in this class: `useState` and `useEffect`. You may read more about these other
+(equally as important but more involved) Hooks [here](https://reactjs.org/docs/hooks-reference.html).
+
+## Hooks
+
+To put it simply, Hooks are functions for use in functional components, making them much more powerful.
+
+You can only call Hooks from the top-level of functional components or your own custom hooks.
+
+Previously, class components were primarily used for complex components in React
+which required their own states and needed to trigger their own side effects based
+on updated values in the component, and more. These functionalities were implemented through
+splitting a component into different parts of its lifecycle, which could easily
+become messy. The introduction of Hooks was a game changer, as it allows developers
+to do away with class-based components and their ugliness.
+
+While Hooks are a new addition to React, their usage along with functional
+components have quickly become a dominant paradigm for React development, allowing
+for developers to reduce boilerplate (code required every time you make a certain feature)
+and make code much more understandable (we've seen how functional programming makes code
+both more concise and less bug-prone), without getting bogged down by confusing
+lifecycle methods.
+
+### `useState`
+
+We've shown the `useState` Hook in previous lectures and you've had to use it
+to keep track of state in A4 and A5. However, let's get deeper into understanding
+it more generally and what you can do with it.
+
+#### Simple Counter Example
+
+Let's consider a simple counter component requiring the use of states. We will
+outline the code in both the old class-based paradigm, and the new functional +
+Hooks paradigm.
+
+##### The Class Component Way
+
+```jsx title="Counter.jsx"
+import React from 'react';
+
+export default class Counter extends React.Component {
+  state = { count: 0 };
+
+  handleClick = () => this.setState({ count: this.state.count + 1 });
+
+  render() {
+    return (
+      <div>
+        <button onClick={this.handleClick}>Click Me!</button>
+        <p>You clicked {this.state.count} times</p>
+      </div>
+    );
+  }
+}
+```
+
+##### The Functional Component + Hooks Way
+
+```jsx title="Counter.jsx / Counter.tsx"
+import React, { useState } from 'react';
+
+export default () => {
```

Comment:
Prefer to use named components and then export it as default rather than anonymous

Reply:
Alright thanks!